portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -41,66 +41,25 @@ public void init() {
     this.commentDataStore = new CommentDataStore(DatastoreServiceFactory.getDatastoreService());
   }
 
-  static class Constants {
-    private final static String PARAM_NAME_COMMENTER = "commenter";
-    private final static String PARAM_NAME_CONTENT = "content";
-  }
-  
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Loads the comment from Datastore
-    List<Comment> comments = getComments();
-
-    // Converts into json form
-    String json = convertToJsonUsingGson(comments);
-
-    // Sends the JSON as the response
-    sendJsonResponse(response, json);
-  }
-
-  /** Loads the comment from Datastore */
-  private List<Comment> getComments() {
-    // Loads comments from Datastore
-    List<Comment> comments = this.commentDataStore.load();
-
-    return comments;
-  }
-
-
-  /** Converts the list of Comment objecct into json form */
-  private String convertToJsonUsingGson(List<Comment> comments) {
-    Gson gson = new Gson();
-    String json = gson.toJson(comments);
-    return json;
-  }
-
-  /** Sends the JSON as response */
-  private void sendJsonResponse(HttpServletResponse response, String json) throws IOException {
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Gets comments from the form
-    String commenter = ServletUtils.getParameter(
-            request, /*name=*/Constants.PARAM_NAME_COMMENTER, /*defaultValue=*/"");
-    String content = ServletUtils.getParameter(
-            request, /*name=*/Constants.PARAM_NAME_CONTENT, /*defaultValue=*/"No comments");
+    String commenter = ServletUtils.getParameter(request, PARAM_NAME_COMMENTER, DEFAULT_COMMENT_COMMENTER);
+    String content = ServletUtils.getParameter(request, PARAM_NAME_CONTENT, DEFAULT_COMMENT_CONTENT);
     // TODO: validate request parameters
 
     // Stores the comment into the Datastore
     storeComment(commenter, content);
 
     // Redirects back to the HTML page.
-    response.sendRedirect("/index.html");
+    response.sendRedirect("/comments.html");
   }
 
   /** Stores the comment into the Datastore */
   private void storeComment(String commenter, String content) {
     // Creates a miscellaneous comment object to convert it to entity
     long timestamp = System.currentTimeMillis();
-    Comment comment = new Comment(commenter, content, timestamp);
+    Comment comment = new Comment(0, commenter, content, timestamp);
```

Comment:
I thought last time we discuss that we don't expose the id for constructor?
I think you only know the id after the data store store that comment into an entity. Optionally you can return the id after storing.

Reply:
Make sense, I will replace by returning id after storing

Reply:
Fixed